### PR TITLE
Allow timestamp strings in 'Table.insert_data'.

### LINKED ISSUE #1382: Support both seconds from epoch and timestamp string
If someone already has a timestamp string, we shouldn't force them to convert it to `datetime.datetime` so that we can in turn [convert that](https://github.com/GoogleCloudPlatform/gcloud-python/blob/0c74a599b6aa0a16e21575998e1770182c05e12c/gcloud/bigquery/table.py#L665-L669) into seconds. (Inspired by #1381.)

From [the docs](https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatype)

> TIMESTAMP data types can be described in two ways: UNIX timestamps or calendar datetimes. BigQuery stores TIMESTAMP data internally as a UNIX timestamp with microsecond precision.

so we could allow the user to just pass a string and we "take their word for it" that it's a valid datetime. Since `Table.insert_data` returns a list of errors anyhow, we wouldn't have to do much/anything to enforce this.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1382.

### CODE REVIEW

File: bigquery/google/cloud/bigquery/table.py [hunk 1]
```diff
@@ -738,15 +738,20 @@ def insert_data(self,
         rows_info = []
         data = {'rows': rows_info}
 
+        def _convert_timestamp(value):
```
Comment (dhermes) [inline bigquery/google/cloud/bigquery/table.py]: Please define this in the global scope

**Comment by dhermes** [comment]
@tseaver Easy enough change to review, ping me once new function is defined as a global

**Comment by tseaver** [comment]
@dhermes ping

**Comment by shealutton** [comment]
Thanks! I needed exactly this change. Much obliged.

**Pull request merged**